sklearn/ensemble/_gb.py

```diff
@@ -28,19 +28,230 @@
 import numpy as np
 from scipy.sparse import csc_matrix, csr_matrix, issparse
 
+from .._loss.loss import (
+    _LOSSES,
+    AbsoluteError,
+    BaseLoss,
+    ExponentialLoss,
+    HalfBinomialLoss,
+    HalfMultinomialLoss,
+    HalfSquaredError,
+    HuberLoss,
+    PinballLoss,
+)
 from ..base import ClassifierMixin, RegressorMixin, _fit_context, is_classifier
+from ..dummy import DummyClassifier, DummyRegressor
 from ..exceptions import NotFittedError
 from ..model_selection import train_test_split
+from ..preprocessing import LabelEncoder
 from ..tree import DecisionTreeRegressor
-from ..tree._tree import DOUBLE, DTYPE
+from ..tree._tree import DOUBLE, DTYPE, TREE_LEAF
 from ..utils import check_array, check_random_state, column_or_1d
 from ..utils._param_validation import HasMethods, Interval, StrOptions
 from ..utils.multiclass import check_classification_targets
+from ..utils.stats import _weighted_percentile
 from ..utils.validation import _check_sample_weight, check_is_fitted
-from . import _gb_losses
 from ._base import BaseEnsemble
 from ._gradient_boosting import _random_sample_mask, predict_stage, predict_stages
 
+_LOSSES = _LOSSES.copy()
+_LOSSES.update(
+    {
+        "quantile": PinballLoss,
+        "huber": HuberLoss,
+    }
+)
+
+
+def _safe_divide(numerator, denominator):
+    """Prevents overflow and division by zero."""
+    with np.errstate(divide="raise"):
+        try:
+            return numerator / denominator
+        except FloatingPointError:
+            return 0.0
```

Comment:
For codecov, can we add a small test to trigger the divide by zero?

Reply:
I tried to find a case with with GB estimator, but that's hard. So I simply added `test_safe_divide`.

Reply:
Apparently, we have a nice example that check this one ;)